Kudu.Core/Scripts/selectNodeVersion.js

```diff
@@ -34,13 +34,26 @@ function resolveNpmPath(npmRootPath, npmVersion) {
 
 
 function getDefaultNpmVersion(nodeVersionPath) {
-    var npmLinkPath = path.resolve(nodeVersionPath, 'npm.txt');
-    // Determine if there's a link to npm at the node path
-    if (!existsSync(npmLinkPath)) {
-        return;
+    var appSettingNpmVersion = process.env.WEBSITE_NPM_DEFAULT_VERSION;
+
+    // extract the current node version from the path
+    var nodePathSplited = nodeVersionPath.split(path.sep).filter(function(e) { return e; });
```

Comment:
I wanted to filter out empty split results. If the path ends in `\` then the last element in the array will be an empty string. I'm not sure if the path here has a trailing slash or not though!